test/contrib/jupyter_notebook_test.py

```diff
@@ -0,0 +1,405 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Tests for `JupyterNotebookTask`.
+
+Requires the following modules: `nbformat`, `nbconvert`, `jupyter`.
+If these modules are not available, some of the tests are skipped.
+"""
+import json
+import logging
+import os
+import tempfile
+import unittest
+
+from luigi import Parameter
+from luigi.local_target import LocalTarget
+
+from luigi.contrib.jupyter_notebook import JupyterNotebookTask
+
+logger = logging.getLogger('luigi-interface')
+
+try:
+    from jupyter_client import KernelManager
+    import nbformat as nbf
+    from nbformat.v4 import (
+        new_notebook,
+        new_code_cell
+    )
```

Comment:
For the final test in this file, you can add these dependencies to `tox.ini` so that the actual test for running a JupyterNotebookTask is confirmed.